Kickstarter-iOS/Views/CommentInputContainerView.swift

```diff
@@ -65,9 +65,26 @@ final class CommentInputContainerView: UIView {
   }
 
   private func setupConstraints() {
+    let postButtonTitleMaximumBoundingRectangle = CGSize(
+      width: .greatestFiniteMagnitude,
+      height: Layout.Button.height
+    )
+    let postButtonTitleConstrainedBoundingRectangle = Strings.Post()
+      .boundingRect(
+        with: postButtonTitleMaximumBoundingRectangle,
+        options: .usesLineFragmentOrigin,
+        attributes: [
+          NSAttributedString
+            .Key.font: self
+            .postButtonTitleFont
+        ],
+        context: nil
+      )
+    let precalculatedPostButtonTitleWidth = ceil(postButtonTitleConstrainedBoundingRectangle.width)
+
     NSLayoutConstraint.activate([
       self.postButton.heightAnchor.constraint(equalToConstant: Layout.Button.height),
-      self.postButton.widthAnchor.constraint(equalToConstant: Layout.Button.width)
+      self.postButton.widthAnchor.constraint(equalToConstant: precalculatedPostButtonTitleWidth)
```

Comment:
You're sure this is the only way to do this? Seems strange that we need to use a fixed width. Is it not because we're making this multiline?

Reply:
Hmm, ok, I didn't check with design or product, but I'll check with them and see if a multiline solution would work as well.

Reply:
No I'm not saying make it multiline, I thought you might have which might have created the need for the fixed width. Let's pair on this a bit later, I would like to see if it's possible (or at least better understand) why the fixed width is needed.

Reply:
Yea for sure.

Reply:
That said - there's wrong with a bit of `String.boundingRect`, it's come in handy a couple times in the past (PillView), but I think here we may just need a `layoutIfNeeded()` or something somewhere 🤔

Reply:
I'll tinker with it a bit more before we pair try and come up with a more elegant solution.

Reply:
Among other things that did not work, I tried adding a spacer view between the `inputTextView` and `postButton` in the `rootStackView` and it does resize the post correctly to the size of the text. However, upon hiding the spacer view when the user starts to type, the post button resizes to fill half the space of the stack view again. Tried with distributions `.fill` and `.fillProportionally`.